src/aws-cpp-sdk-core/include/smithy/client/AwsSmithyClientBase.h

```diff
@@ -55,6 +55,12 @@ namespace Aws
     namespace Endpoint
     {
         class AWSEndpoint;
+
+        struct AWSEndpointResolutionOverrides
```

Comment:
there is a `EndpointUpdateCallback&& endpointCallback,` to insert the path segments and set this boolean variable within the generated code